### ISSUE TITLE
[Breaking] Change weights generation to follow WeightInitializers.jl

### ISSUE THREAD

**Issue by MartinuzziFrancesco** [SciML/ReservoirComputing.jl#184]
In order to streamline the package and creation of reservoir computing models the creation of the weights of the matrices should follow [WeightInitializers.jl](https://github.com/LuxDL/WeightInitializers.jl)

**Comment by Jay-sanjay** [comment]
Hi @MartinuzziFrancesco  I guess I can take a look at this, if you can elaborate a bit what exactly needs to be done here.

**Comment by MartinuzziFrancesco** [comment]
Hi, of course! So at the moment the construction of the input layers matrices and reservoir matrices uses abstract typing and dispatch over different `struct`s of the different architectures for the matrices and it is a bit cumbersome. I wanted to simplify it a bit and make it follow the standard in WeightInitializers.jl. This way we would also start an easier way to control the randomness in ReservoirComputing.jl. For example `SparseRandReservoir` would look like this:

```julia
using ReservoirComputing, Random
rng = MersenneTwister(42)

# Passing kwargs with explicit rng call
res_cl = rand_sparse(rng; sparsity=0.1, spectral_radius=1.0)
#res_cl is now a callable function that takes in the size of the reservoir
reservoir_weights = res_cl(rng, 10, 10)

# Passing kwargs with default rng call
res_cl = rand_sparse(; sparsity=0.1, spectral_radius=1.0)
#res_cl is now a callable function that takes in the size of the reservoir
reservoir_weights = res_cl(10, 10)
```

For details of the implementation you can refer to the source in WeightInitializers. I could sketch `rand_sparse` during this week if that would be of help. This needs to be done for both [input layers](https://github.com/SciML/ReservoirComputing.jl/blob/master/src/esn/esn_input_layers.jl) and [reservoirs](https://github.com/SciML/ReservoirComputing.jl/blob/master/src/esn/esn_reservoirs.jl). 

Of course after this we need to modify the `ESN` constructor but that should be rather straightforward.

This issue is part of a streamlining in order to be more in line with (F)Lux as model building philosophy. This should also make contributions a little bit easier

**Comment by MartinuzziFrancesco** [comment]
@Jay-sanjay I have started a PR in #193, you can follow the template to create the remaining functions for the reservoir and input layers. I will need to think about a couple of details (and tests as well)

**Comment by Jay-sanjay** [comment]
Thanks @MartinuzziFrancesco  for the template, I will start to work on it with other functions

**Comment by MartinuzziFrancesco** [comment]
Addressed in #196 and finalized in #203
